### PR TITLE
Interaction state hover for checkbox

### LINKED ISSUE #2090: Interaction State Hover for Checkbox
Requires custom implementation.

Utilize `--background-checked`, `--background` etc. CSS custom properties.

See https://ionicframework.com/docs/v5/api/checkbox

### PR DESCRIPTION
**Pull request by jkaltoft**
## Which issue does this PR close?

This PR closes #2090 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [x] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/stable/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be automatically merged to `stable` via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

File: libs/designsystem/src/lib/components/checkbox/checkbox.component.scss [hunk 1]
```diff
@@ -12,6 +13,23 @@ $default-checkbox-radio-size: map.get(utils.$checkbox-radio-sizes, 'md');
...
+    $_lightness: utils.get-loudness($loudness);
+  } @else {
+    // negative number means "make it darker"
+    $_lightness: -1 * utils.get-loudness($loudness);
+  }
+
+  $hover-color: #{color.scale($_background, $lightness: $_lightness)};
+
+  @return $hover-color;
+}
```
Comment (jkaltoft) [inline libs/designsystem/src/lib/components/checkbox/checkbox.component.scss:31]: I'm thinking, maybe this should be generalised and moved to the shared interaction state module.
What does the reviewer think?

Comment (RasmusKjeldgaard) [reply libs/designsystem/src/lib/components/checkbox/checkbox.component.scss:31]: Since it is used in #2128 I fully agree!

**Review by RasmusKjeldgaard — APPROVED** [review]

File: libs/designsystem/src/lib/components/checkbox/checkbox.component.scss [hunk 2]
```diff
@@ -20,6 +38,10 @@ $default-checkbox-radio-size: map.get(utils.$checkbox-radio-sizes, 'md');
     --background-checked: #{utils.get-color('success')};
     --border-color-checked: #{utils.get-color('success')};
 
+    @include interaction-state.apply-hover {
+      --background-checked: #{_get-hover-color('success')};
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/src/lib/components/checkbox/checkbox.component.scss:42]: Should this have loudness xxl or similar too? It is very hard to see the hover effect when it is checked with success color.

Comment (jkaltoft) [reply libs/designsystem/src/lib/components/checkbox/checkbox.component.scss:42]: The request from UX in Zeplin is to use the default loudness. I agree that it's very hard to see, but maybe that discussion should be saved for the UX review?

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**